src/util/bson/bson.cpp

```diff
@@ -829,8 +832,10 @@ bool Parser::string(string_t& val) {
             m_instructions.push({State::Skip});
             break;
         case State::Date: {
-            auto epoch = atol(val.data());
-            m_marks.top().push_back(instruction.key, Datetime(epoch));
+            int64_t millis_since_epoch = atoll(val.data());
+            auto ts = realm::Timestamp(millis_since_epoch/1000,
```

Comment:
This isn't correct for negative `millis_since_epoch`. Consider when `millis_since_epoch` is -1005. This logic would give `{-1, -5}`, when you really want `{-2, 995}`.

Reply:
Looks like this is actually the style of input that `realm::Timestamp` requires.